config/unicorn.conf.rb

```diff
@@ -144,25 +144,33 @@ def check_sidekiq_heartbeat
           @sidekiq_next_heartbeat_check ||= Time.new.to_i + @sidekiq_heartbeat_interval
 
           if @sidekiq_next_heartbeat_check < Time.new.to_i
-
-            last_heartbeat = Jobs::RunHeartbeat.last_heartbeat
-            restart = false
+            @sidekiq_next_heartbeat_check = Time.new.to_i + @sidekiq_heartbeat_interval
+            restarted = false
 
             if out_of_memory?
               Rails.logger.warn("Sidekiq is consuming too much memory (using: %0.2fM) for '%s', restarting" % [(max_rss.to_f / 1.megabyte), ENV["DISCOURSE_HOSTNAME"]])
-              restart = true
+              Demon::Sidekiq.restart
+              restarted = true
             end
 
-            if last_heartbeat < Time.new.to_i - @sidekiq_heartbeat_interval
-              STDERR.puts "Sidekiq heartbeat test failed, restarting"
-              Rails.logger.warn "Sidekiq heartbeat test failed, restarting"
-
-              restart = true
+            if !restarted
+              Demon::Sidekiq.heartbeat_queues.each do |queue|
```

Comment:
hmmm is this checking all heartbeat queues? Should it not just check heartbeats for child sidekiq processes it has?

Reply:
`Demon::Sidekiq.heartbeat_queues` returns all sidekiq processes that are spawned by unicorn master and we check their heartbeat one by one here. Is this not what we supposed to do?

Reply:
Oh I see... so each process will give a different list here ...

A confusing thing here is that if you call `Demon::Sidekiq.heartbeat_queues` you will get one list .. and if you call this on a forked child process you will get an empty list. Maybe we call this `Demon::Sidekiq.child_process_heartbeat_queues` ?